apps/browser/src/autofill/services/autofill.service.ts

```
@@ -106,12 +115,22 @@ export default class AutofillService implements AutofillServiceInterface {
         onlyVisibleFields: options.onlyVisibleFields || false,
         fillNewPassword: options.fillNewPassword || false,
         cipher: options.cipher,
+        tabUrl: tab.url,
       });
 
       if (!fillScript || !fillScript.script || !fillScript.script.length) {
         return;
       }
 
+      if (
+        fillScript.untrustedIframe &&
+        options.allowUntrustedIframe != undefined &&
+        !options.allowUntrustedIframe
+      ) {
+        this.logService.info("Auto-fill on page load was blocked due to an untrusted iframe.");
+        return;
```

I had thought we were going to display a popup to the user. This would fail silently to most people.

---

This is only hit for autofill on page load case, which I agree that we should fail on.